cli/src/commands/extensions/extension.rs

```diff
@@ -104,12 +93,47 @@ impl Extension {
                 );
             } else if source_path.is_file() {
                 if dest_path.exists() {
-                    return Err(anyhow!("{}: already exists", dest_path.to_string_lossy()));
+                    return Err(anyhow!("{}: already exists", dest_path.display()));
                 } else {
                     fs::copy(source_path, dest_path)?;
                 }
             }
         }
+        Ok(())
+    }
+
+    /// Install the extension in the default path.
+    pub fn install(&self) -> Result<()> {
+        println!("Installing extension {}...", self.name());
+        let target_prefix = extension_path(self.name())?;
+
+        if target_prefix == self.path {
+            return Err(anyhow!("extension path and installation path are identical, skipping"));
+        }
+
+        if target_prefix.exists() {
+            if dir_compare(&self.path, &target_prefix)? {
+                return Err(anyhow!("identical extension already installed, skipping"));
+            } else {
+                let mut prompt = Confirm::new();
+                prompt
+                    .with_prompt(format!(
+                        "Another version of the '{}' extension is already installed. Overwrite?",
+                        self.name()
+                    ))
+                    .default(true);
```

Comment:
I would cut down on complexity, remove this check and reword to something like `The '{}' extension is already installed. Overwrite?`.

IIUC, not having it would only mean that we overwrite files that are already identical, which is a very modest overhead unless we anticipate cli extensions to be in the hundreds of MB of size -- and even then, it is much more a networking issue rather than disk writes issue. Having it has a cost in term of complexity.

Do we need the `dir_compare` check, or to report to the user that the installed extension is identical? Are there other reasons/places it could be useful? If so, I'd also add a few tests with fixtures for it.

Reply:
> Do we need the `dir_compare` check, or to report to the user that the installed extension is identical? Are there other reasons/places it could be useful?

The `dir_compare` check is used to avoid excessive prompting during the following scenario: The `phylum extension install` command will be used during `phylum update` to install or upgrade official extensions. Since extensions may not change between every release, we want to avoid prompting for every official extension during every `phylum update`.

Reply:
> If so, I'd also add a few tests with fixtures for it.

I'll have to think a bit more on this. Testing the interactive prompt is difficult and we avoid it in other cases (e.g., the permissions prompt). But I guess I could add a test case to confirm that install is aborted when attempting to install a modified version of an extension in a non-interactive setting.

Reply:
> The `dir_compare` check is used to avoid excessive prompting during the following scenario: The `phylum extension install` command will be used during `phylum update` to install or upgrade official extensions. Since extensions may not change between every release, we want to avoid prompting for every official extension during every `phylum update`.

In that case, yeah, the check is a good thing.

> I'll have to think a bit more on this. Testing the interactive prompt is difficult and we avoid it in other cases (e.g., the permissions prompt). But I guess I could add a test case to confirm that install is aborted when attempting to install a modified version of an extension in a non-interactive setting.

I'd say it wouldn't be necessary to go that far, though an e2e test of this feature is welcome.

What I'd like to mainly have more coverage on is just `dir_compare` and `file_compare`, and we could do that with no prompting at all, by just having tests that create, populate and compare some temporary directories and files (respectively).

Reply:
Good point. I was thinking e2e test, but unit testing is much simpler. I will add unit tests.